app/controllers/declaration/AdditionalDeclarationTypePageController.scala

```diff
@@ -61,13 +59,25 @@ class AdditionalDeclarationTypePageController @Inject()(
       .fold(
         (formWithErrors: Form[AdditionalDeclarationType]) =>
           Future.successful(BadRequest(declaration_type(formWithErrors))),
-        validAdditionalDeclarationType =>
+        validAdditionalDeclarationType => {
+          request.authenticatedRequest.session.data.get("sessionId")
+            .map(updateCache(_, validAdditionalDeclarationType))
+
           customsCacheService
             .cache[AdditionalDeclarationType](cacheId, decType.formId, validAdditionalDeclarationType)
             .map(_ => Redirect(controllers.declaration.routes.ConsignmentReferencesController.displayPage()))
+        }
       )
   }
 
+  private def updateCache(sessionId: String, formData: AdditionalDeclarationType): Future[Option[Future[Either[String, JourneyCacheModel]]]] = {
```

Comment:
@PatrykRudnicki @ed0906 @benjaminparker is there a way of making this method `generic`, so that we do not need to  repeat ourselves. 

This is the line that would be nice to generalise 
```
model.copy(additionalDeclarationType = Some(formData))
```

Reply:
We have made this more generic by adding ModelCacheable Trait